libraries/HumanPoseFuser.js

```diff
@@ -0,0 +1,803 @@
+        let poseConfidence = this.computeOverallConfidence(wholePose.joints);
+        wholePose.poseConfidence = poseConfidence;
+        Object.defineProperty(wholePose, 'poseConfidence', { enumerable: false });
+
+        this.pastPoses[objectId].poses.push(wholePose);
+    }
+
+    /** Computes overall confidence of whole pose from joint confidences.
+     * @param {Array.< {x: number, y: number, z: number, confidence: number} >} poseJoints
+     * @returns {number} overall pose confidence
+     */
+    computeOverallConfidence(poseJoints) {
+
+        if (poseJoints.length == 0) {
+            return 0.0;
+        }
+
+        // limit to joints which are not synthetically computed,
+        // and keep just one 'head' joint - nose to limit an influence of head
+        const selectedJointNames = ['nose', 'left_shoulder', 'right_shoulder', 'left_elbow', 'right_elbow', 'left_wrist', 'right_wrist', 'left_hip', 'right_hip', 'left_knee', 'right_knee', 'left_ankle', 'right_ankle'];
+
+        const joints = Object.values(JOINTS);
+        let sum = 0.0;
+        for (let name of selectedJointNames) {
+            const jointIndex = joints.indexOf(name);
+
+            if (jointIndex >= 0) {
+                sum += poseJoints[jointIndex].confidence;
+            }
+        }
+
+        return sum / selectedJointNames.length;
+    }
+
+    /** Removes internal data about a given human object
+     * @param {string} objectId - identificator of human object
+     */
+    removeHumanObject(objectId) {
+
+        if (this.verbose) {
+            console.log('removing human obj=' + objectId);
+        }
+
+        // remove pose history of a deleted human object if there is any
+        delete this.pastPoses[objectId];
+
+        if (this.humanObjectsOfFusedObject[objectId] === undefined) { // a standard human object
+            let fusedObjectId = this.humanObjectsOfFusedObject.getFusedObject(objectId);
+
+            // remove association of a deleted object to any fused human object
+            this.humanObjectsOfFusedObject.deleteObject(objectId);
+
+            // remove fused human objects if it has less than 2 associated human objects
+            if (fusedObjectId && this.humanObjectsOfFusedObject[fusedObjectId].length < 2) {
+                this.removeHumanObject(fusedObjectId);
+            }
+
+        } else {  // a fused human object
+            // remove parent reference from its remaining associated human objects
+            for (let id of this.humanObjectsOfFusedObject[objectId]) {
+                if (this.objectsRef[id] !== undefined) {
+                    this.objectsRef[id].parent = null;
+
+                    this.batchedUpdates[id] = {
+                        objectKey: id,
+                        frameKey: null,
+                        nodeKey: null,
+                        propertyPath: 'parent',
+                        newValue: null,
+                        editorId: 0    // TODO: some server identificator
```

Comment:
Yeah that seems about right. We do have access to the utilities.uuidTime() function which will produce a random string, but we would need a modified version of the function to seed it with the server IP instead of the current timestamp, if we want it to return a consistent result per server

Reply:
It should be technically correct with the current implementation, though, as the editorId is mainly used to prevent sending messages back to the toolbox app that originated the message. But since this message is coming from the server it doesn't need to be filtered this way.

Reply:
Right. I did not encounter any issues with this so far. I will keep it as is.